File: src/main/java/Fibonacci.java
import java.util.InputMismatchException;
import java.util.Scanner;

public class Fibonacci {
    public static void main(String[] args) {
        boolean exit = false;
        do{
            Scanner scanner = new Scanner(System.in);
            long n = 0;
            System.out.println("Introduzca un número: ");
            try{
                long startTime = System.nanoTime();

                n = scanner.nextLong();
                System.out.println("El fibonacci de " + n + " es: " + fibonnacciRecursivo(n));
                exit = true;

                long stopTime = System.nanoTime();
                long elapsedTime = (stopTime - startTime)/1_000_000_000;
                System.out.println("Time : " + elapsedTime + " seconds");

            }catch (InputMismatchException ime){
                System.out.println("Debe de introducir un número");
            }
        }while (!exit);
    }


    static long fibonnacciRecursivo(long n){
    if(n <= 1)
        return n;
    return (fibonnacciRecursivo(n - 1) + fibonnacciRecursivo(n - 2));
    }
}

File: Go/fibonacci.go
package main

import "fmt"
import "bufio"
import "strconv"
import "os"
import "time"

func main() {

  start := time.Now()

   exit := false
  for{
    scanner := bufio.NewScanner(os.Stdin)
    fmt.Println("Introduzca un número: ")
    scanner.Scan()

    if n,error := strconv.Atoi(scanner.Text()); error == nil{
      fmt.Println("El Fibonacci de " + strconv.Itoa(n)+" es " + strconv.Itoa(fib(n)))
      exit = true
    }else{
      fmt.Println("Debe de insertar un número")
    }
    if exit{
      break
    }
  }
elapsed:=time.Since(start)
fmt.Printf("Time : %s \n",elapsed)
}

func fib(n int) int{
  if n <= 1{
    return n
  }
  return fib(n - 1) + fib(n - 2)
}
